generate-task-list.js

```diff
@@ -0,0 +1,59 @@
+function toCamelCase(text) {
+	return text.replace(/^([A-Z])|[\s-_]+(\w)/g, function (match, p1, p2, offset) {
+		if (p2) return p2.toUpperCase();
+		return p1.toLowerCase();
+	});
+}
+
+function camelCaseToString(str, separator) {
+	separator = typeof separator === 'undefined' ? '_' : separator;
+
+	return str
+		.replace(/([a-z\d])([A-Z])/g, '$1' + separator + '$2')
+		.replace(/([A-Z]+)([A-Z][a-z\d]+)/g, '$1' + separator + '$2')
+		.toLowerCase();
+}
+
+const addTaskForm = document.querySelector('.task-form__add');
+const taskList = document.querySelector('.task-list');
+const addTask = event => {
+	event.preventDefault();
+
+	const taskName = document.querySelector('task-form__input', 'task-form__input--task-content').value;
+	const taskDeadline = document.querySelector('task-form__input', 'task-form__input--date').value;
+
+	let lastElement = tasks.pop();
```

Comment:
`tasks.pop` does something different that you might have thought - it not only returns the last element from the `tasks` array, it also **removes** it.

```
const items = [1, 2, 3];
const lastElement = items.pop(); // 3
console.log(items); // [1, 2]
```

We say that `.pop()` method mutates the array that it's performed on. As you can see, there are some hard to notice consequences that mutation carries.

In bigger projects you can use some e.g Ramda helper, but retrieving the last in vanilla JS can look like this:
```
const items = [11, 22, 33];
const lastElementIndex = items.length - 1; // 2
const lastElement = items[lastEllementIndex]; // 33
console.log(items); // [11, 22, 33];
```